src/table/virtualized-table/Header.tsx

```diff
@@ -6,8 +6,9 @@ import { useContextSelector, TableContext } from '../context';
 import useResetHeader from './hooks/use-reset-header';
 
 const Header = (props: HeaderProps) => {
-  const { rect, forwardRef, columns, columnWidths, pageInfo, headerStyle, rowHeight } = props;
+  const { rect, forwardRef, columns, pageInfo, headerStyle, rowHeight } = props;
   const { layout } = useContextSelector(TableContext, (value) => value.baseProps);
+  const columnWidths = useContextSelector(TableContext, (value) => value.columnWidths);
```

Comment:
may be silly question but, why not getting both at the same call to `useContextSelector`? is it something about re-rendering? if so, do we need to care about it at this level?

Reply:
I do think that because of how `useContextSelector` works. It triggers a re-render that is tighly coupled with what the callback function returns. So every time the callback function returns a new value. The component is re-rendered.

So you cannot really get both values in the same call, as something like `const { columnWidths, baseProps } = useContextSelector(TableContext, (value) => value);` would always return trigger a re-render as `value` is always a new object on every render.

If that is what you are asking?

Reply:
Yes, that makes sense then 👍